chunky/src/java/se/llbit/math/bvh/MidpointBVH.java

```diff
@@ -0,0 +1,153 @@
+/* Copyright (c) 2021 Jesper Öqvist <jesper@llbit.se>
+ * Copyright (c) 2021 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.math.bvh;
+
+import se.llbit.chunky.entity.Entity;
+import se.llbit.chunky.main.Chunky;
+import se.llbit.log.Log;
+import se.llbit.math.AABB;
+import se.llbit.math.Vector3;
+import se.llbit.math.primitive.Primitive;
+import se.llbit.util.TaskTracker;
+
+import java.util.*;
+import java.util.function.IntConsumer;
+
+public class MidpointBVH extends BinaryBVH {
+    public static void initImplementation() {
```

Comment:
```suggestion
    public static void register() {
```

(nit) maybe? (to make clear what this does)?